### PR TITLE
Change logic for finding plugin slug in CLI

### LINKED ISSUE #364: Improve parsing of associative arguments to accept arguments regardless of position
Command 1:

    wp plugin check hello.php --fields=line,column,type,code

Command 2:

    wp plugin check --fields=line,column,type,code hello.php

These both commands are expected to give same output. But it seems our CLI breaks when assoc args are passed before the `plugin` slug parameter.

PHP fatal error:

    Fatal error: Uncaught Exception: Invalid plugin slug: Plugin with slug --fields=line,column,type,code is not installed. in /var/www/html/wp-content/plugins/plugin-check/includes/Utilities/Plugin_Request_Utility.php:77

### PR DESCRIPTION
**Pull request by ernilambar**
Fixes https://github.com/WordPress/plugin-check/issues/364

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

File: phpstan.neon.dist [hunk 1]
```diff
@@ -21,3 +21,7 @@ parameters:
         - includes/Checker/Checks/Abstract_File_Check.php
         - includes/Traits/Find_Readme.php
         - includes/Traits/File_Editor_URL.php
+    -
+      message: '/^Function str_starts_with not found.$/'
+      paths:
+        - includes/Checker/CLI_Runner.php
```
Comment (mukeshpanchal27) [inline phpstan.neon.dist]:
```suggestion
      path: includes/Checker/CLI_Runner.php
```

For single file let's use `path`

File: includes/Checker/CLI_Runner.php [hunk 2]
```diff
@@ -57,8 +57,19 @@ public static function is_plugin_check() {
...
 	protected function get_plugin_param() {
-		// Get the plugin parameter from the command line arguments.
-		$plugin = isset( $_SERVER['argv'][3] ) ? $_SERVER['argv'][3] : '';
+		// Exclude first three reserved elements.
+		$params = array_slice( $_SERVER['argv'], 3 );
+
+		// Remove associative arguments.
+		$params = array_filter(
+			$params,
+			function ( $val ) {
```
Comment (mukeshpanchal27) [inline includes/Checker/CLI_Runner.php]:
```suggestion
			static function ( $val ) {
```

Per [729](https://github.com/WordPress/performance/pull/729), Use static closures for minor performance improvement whenever instance access is not needed.

**Review by mukeshpanchal27 — COMMENTED** [review]
Thank you, @ernilambar, for the PR! I've provided some feedbacks. Aside from that, everything appears to be in good from my perspective.

**Comment by ernilambar** [comment]
PR updated as per the review comments.

**Comment by ernilambar** [comment]
@mukeshpanchal27 After last commit, Behat test (PHP 8.0 WP Latest) is failing. Any idea?

**Comment by swissspidy** [comment]
Does it pass locally for you? There might be some error/notice happening

**Comment by ernilambar** [comment]
> Does it pass locally for you? There might be some error/notice happening

I could not make it run Behat tests locally. Some database error occurs and tests fails even before starting. So I usually check those in GitHub Actions.

**Comment by mukeshpanchal27** [comment]
> @mukeshpanchal27 After last commit, Behat test (PHP 8.0 WP Latest) is failing. Any idea?

The [nikic/php-parser](https://github.com/nikic/PHP-Parser/releases/tag/v5.0.0) get updated from 4.18.0 to 5.0.0 in past few hours that may be issue as they update things related to PHP versions. cc. @swissspidy

**Comment by swissspidy** [comment]
> > @mukeshpanchal27 After last commit, Behat test (PHP 8.0 WP Latest) is failing. Any idea?
> 
> The [nikic/php-parser](https://github.com/nikic/PHP-Parser/releases/tag/v5.0.0) get updated from 4.18.0 to 5.0.0 in past few hours that may be issue as they update things related to PHP versions. cc. @swissspidy

This project dos not use PHP-Parser, nor do any of the dependencies.

> I could not make it run Behat tests locally. Some database error occurs and tests fails even before starting. So I usually check those in GitHub Actions.

Have you tried running `composer prepare-behat-tests` first to set up the database? Or run `WP_CLI_TEST_DBTYPE=sqlite composer behat` to test without MySQL.

-----

From what I can see [in the logs](https://github.com/WordPress/plugin-check/actions/workflows/behat-test.yml), this failure started already at https://github.com/WordPress/plugin-check/actions/runs/7443210203 (though it worked fine on the PR branch itself)

So it's not isolated to this PR.

Unfortunately I'm unable to reproduce this locally right now, so let's deal with this in a separate PR to debug in more detail. No blocker for this PR.

**Comment by mukeshpanchal27** [comment]
> Have you tried running `composer prepare-behat-tests` first to set up the database? Or run `WP_CLI_TEST_DBTYPE=sqlite composer behat` to test without MySQL.

I've tested it locally, and it seems to be working fine for me as well. Not entirely sure why the issue persists. I've opened a dedicated issue to track and investigate that. Thank you!

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**